axelor-human-resource/src/main/java/com/axelor/apps/hr/service/timesheet/TimesheetServiceImpl.java

```diff
@@ -1218,6 +1223,97 @@ protected TimesheetLine createTimeSheetLineFromPPT(
     return timesheetLine;
   }
 
+  protected BigDecimal getConvertedPlannedTime(
+      Timesheet timesheet, ProjectPlanningTime projectPlanningTime) throws AxelorException {
+
+    String timeLoggingPreference = timesheet.getTimeLoggingPreferenceSelect();
+
+    if (timeLoggingPreference == null) {
+      timeLoggingPreference = timesheet.getEmployee().getTimeLoggingPreferenceSelect();
+    }
+
+    switch (timeLoggingPreference) {
```

Comment:
if timelogging preference is empty on timesheet and on employee, we get a NPE. Please throw an error instead.